src/zstr.hpp

```diff
@@ -420,9 +417,9 @@ class ifstream
       public std::istream
 {
 public:
-    explicit ifstream(const std::string& filename, std::ios_base::openmode mode = std::ios_base::in)
+    explicit ifstream(const std::string filename, std::ios_base::openmode mode = std::ios_base::in, size_t buff_size = default_buff_size)
```

Comment:
Why do you remove the ampersand in filename?

Reply:
This was to facilitate things like: zstr::ifstream mygz(filename + ".gz"); In practice, the compiler will make it equivalent to a reference argument wherever possible but still allows such flexibility to the programmer.

Reply:
oh I see. But shouldn't we then make it in all function arguments where "filename" appears, like open, so that it's consistent?

Reply:
Sure, truth be told this slipped into this branch because I'm working on a different branch where I wanted it, but yes I think leaving the reference off is more flexible.
 b55f0ecd672 contains that fix.